filer/admin/folderadmin.py

```diff
@@ -1169,8 +1169,21 @@ def _list_all_to_resize(self, request, files_queryset, folders_queryset):
         return to_resize
 
     def _new_subject_location(self, original_width, original_height, new_width, new_height, x, y, crop):
-        # TODO: We could probably do better
-        return (round(new_width / 2), round(new_height / 2))
+        # TODO: We could probably do even better, but this method knows nothing
+        # about actual thumbnailing algorithm details.
+        # It's better to reset subject location to the central point of the new
+        # image if any of these two conditions are met:
+        # - 'scale_and_crop_with_subject_location' thumbnail processor is
+        # disabled. Resizing would not respect subject location.
+        # - The image was cropped. The originally specified subject location
+        # is either near the new center (if subject location was respected) or
+        # could be outside of the new image.
+        if crop or 'filer.thumbnail_processors.scale_and_crop_with_subject_location' not in django_settings.THUMBNAIL_PROCESSORS:
```

Comment:
You're probably right. I've adjusted this condition and tests (just doing the same assertions for both original and subject-aware thumbnail processors now).